vlib/x/vweb/README.md

```diff
@@ -0,0 +1,770 @@
+
+Now modifying any file in your web app (whether it's a .v file with the backend logic
+or a compiled .html template file) will result in an instant refresh of your app
+in the browser. No need to quit the app, rebuild it, and refresh the page in the browser!
+
+Now modifying any file in your web app (whether it's a .v file with the backend logic
+or a compiled .html template file) will result in an instant refresh of your app
+in the browser. No need to quit the app, rebuild it, and refresh the page in the browser!
+
+## Deploying vweb apps
+
+All the code, including HTML templates, is in one binary file. That's all you need to deploy.
+Use the `-prod` flag when building for production.
+
+## Getting Started
+
+To start, you have to import the module `x.vweb` and define a structure that will 
+represent your app and a structure which will represent the context of a request.
+These structures have to be declared with the `pub` keyword.
+
+**Example:**
+```v
+module main
+
+import x.vweb
+
+// Our context struct must embed `vweb.Context`!
+pub struct Context {
+	vweb.Context
+}
+
+pub struct App {
+	name string
+}
+
+fn main() {
+	mut app := &App{}
+	// Pass the App and context type and start the web server on port 8080
+	vweb.run[App, Context](mut app, 8080)
+}
+```
+
+You can use the `App` struct for data you want to keep during the lifetime of your program,
+or for data that you want to share between different routes.
+
+The `Context` struct is created every time a request is received, so you could use this struct
+for data that could be different for each request.
+
+## Defining endpoints
+
+To add endpoints to your web server, you have to extend the `App` struct.
+For routing you can either use auto-mapping of function names or specify the path as an attribute.
+The function expects a parameter of your Context type and a response of the type `vweb.Result`.
+
+**Example:**
+```v ignore
+// This endpoint can be accessed via http://localhost:port/hello
+pub fn (app &App) hello(mut ctx Context) vweb.Result {
+	return ctx.text('Hello')
+}
+
+// This endpoint can be accessed via http://localhost:port/foo
+@['/foo']
+pub fn (app &App) world(mut ctx Context) vweb.Result {
+	return ctx.text('World')
+}
+```
+
+### HTTP verbs
+
+To use any HTTP verbs (or methods, as they are properly called),
+such as `@[post]`, `@[get]`, `@[put]`, `@[patch]` or `@[delete]`
+you can simply add the attribute before the function definition.
+
+**Example:**
+
+```v ignore
+// only GET requests to http://localhost:port/world are handled by this method
+@[get]
+pub fn (app &App) world(mut ctx Context) vweb.Result {
+	return ctx.text('World')
+}
+
+// only POST requests to http://localhost:port/product/create are handled by this method
+@['/product/create'; post]
+pub fn (app &App) create_product(mut ctx Context) vweb.Result {
+	return ctx.text('product')
+}
+```
+
+### Routes with Parameters
+
+Parameters are passed directly to an endpoint route using the colon sign `:`
```

Comment:
Actually, unless it's changed in v2, parameters are passed as the type they are declared in the function.  They are only pass as string if the parameter type is `string`.

Reply:
Wow did not know this! Will check if it's in v's compiler code or vweb itself